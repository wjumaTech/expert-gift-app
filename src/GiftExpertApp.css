body {
  background-color: #F0F0F0;
}

h1,
h2,
h3,
p,
.card-text,
.card-title {
  color: #213555 !important;

}

.card-body {
  background-color: #E5D283;
}

.form-control,
.form-control:active,
.form-control:focus {
  background-color: #E5D283;
  box-shadow: none;
}

.form-control:focus {
  border: 1px solid #213555;
}

.card {
  transition: 0.3s ease-in;
}

.card:hover {
  border: 1px solid #213555;
  cursor: pointer;
  box-shadow: 0 3px 3px 0;
  border-radius: 0;
}