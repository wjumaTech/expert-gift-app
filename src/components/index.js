export * from './add-category/add-category.component';
export * from './gift-grid/gift-grid.component';