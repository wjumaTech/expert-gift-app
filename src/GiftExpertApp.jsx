import { useState } from "react";

import { AddCategory } from "./components/add-category/add-category.component";
import { GiftGrid } from './components/gift-grid/gift-grid.component';

import './GiftExpertApp.css';


export const GiftExppertApp = () => {

  const [categories, setCategories] = useState(['One Punch']);

  const onAddCategory = (newCategory) => {
    newCategory = newCategory.toLowerCase();
    if (categories.includes(newCategory)) return;
    setCategories([...categories, newCategory]);
  };

  return (
    <div className="container mt-5">
      <h1>Gift Expert App</h1>
      <hr />
      <AddCategory 
        onNewCategory={ onAddCategory }
      />
      {
        categories.map(category => (<GiftGrid key={category} category={category} />))
      }
    </div>);
};