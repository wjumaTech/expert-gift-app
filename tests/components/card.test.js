import { render, screen } from '@testing-library/react';
import { Card } from '../../src/components';


let id = 'asaddrwdfsdsd';
let title = 'saitama';
let image = 'https://one-punch/saitama.jpg';


describe('Probando el CARD COMPONENT', () => {

	test('Debe de hacer match con el SNAPSHOT', () => {

		const { container } = render(<Card id={id} title={title} image={image} />);
		expect(container).toMatchSnapshot();

	});


	test('Debe de mostrar la imagen con el src y el alt valido', () => {

		// Si necesitaramos ver el container lo hariamos de la siguiente manera
		render(<Card id={id} title={title} image={image} />);
		// screen.debug();

		// expect( screen.getByRole('img').src ).toBe( image );

		const { src, alt } = screen.getByRole('img');
		expect(src).toBe(image);
		expect(alt).toBe(title);

	});

	test('Debe de mostrar el titulo en el componente', () => {

		render(<Card id={id} title={title} image={image} />);
		// screen.debug();
		expect( screen.getByText(title) ).toBeTruthy();

	});

});