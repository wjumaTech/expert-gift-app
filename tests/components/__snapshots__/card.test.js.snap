// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando el CARD COMPONENT Debe de hacer match con el SNAPSHOT 1`] = `
<div>
  <div
    class="card mb-3"
  >
    <img
      alt="saitama"
      src="https://one-punch/saitama.jpg"
    />
    <div
      class="card-body"
    >
      <h3
        class="card-title"
      >
        saitama
      </h3>
      <p
        class="card-text"
      >
        asaddrwdfsdsd
      </p>
    </div>
  </div>
</div>
`;
